src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -159,7 +166,14 @@ public async Task<string> GetBrokerAuthTokenSilentlyAsync(IDictionary<string, st
 
         public void GetBrokerAccountInfo(IDictionary<string, string> brokerPayload, Activity callerActivity)
         {
-            var loginHint = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint);
+            string loginHint = string.Empty;
+            if (string.IsNullOrEmpty(GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint)))
+            {
+                _logger.Error(MsalErrorMessage.NoUPNOrAccountIDForSilentAuth);
+                throw new MsalClientException(MsalError.NoUsernameOrAccountIDProvidedForSilentAndroidBrokerAuthentication, MsalErrorMessage.NoUPNOrAccountIDForSilentAuth);
```

Comment:
@bgavrilMS it should not be possible to have both the username and home account id to be null correct? what do you think i should throw here

Reply:
Yes, that is a correct. ATS requies the user to pass in a login hint or an IAccount.